Twig Templates Manual (#1176)
contao/docs: docs/manual/layout/templates/twig/twig-syntax.en.md

---
title: "Twig syntax"
description: "Syntax of Twig."
url: "layout/templates/twig/syntax"
weight: 20
---

Twig templates have their own syntax. We present here only the most important rules necessary for a basic understanding
of Twig.

{{% notice info %}}
The Twig syntax is [well documented](https://twig.symfony.com/doc/3.x/). As a starting point the
section [Twig for template designers](https://twig.symfony.com/doc/3.x/templates.html) is recommended.
{{% /notice %}}

## Identifier

In Twig, the following three identifiers are used.

* `{# ... #}` - [Comments](#comments)

> * `{{ ... }}` - [Output of variable](#output-of-variables)

Reviewer: ```suggestion
* `{{ ... }}` - [Output of a variable](#output-of-variables)
```

* `{% ... %}` - [Commands and control structures](#commands-and-control-structures)

### Comments

A comment can be single-line or multi-line. Everything between `{#` and `#}` is commented out.

{{% example "Single-line comment" %}}
```twig
{# my comment #}
```
{{% /example %}}

It is also possible to comment out parts of the code.

{{% example "Multiline comment with commented out code" %}}
```twig
{# commented out code - the code will not be executed
{{ variable }}
#}
```
{{% /example %}}

### Output of variables

You can output a variable with `{{ name_of_variable }}`.

{{% example "Output of a variable" %}}
```twig
<p>output: {{ name_of_variable }} </p>
```
{{% /example %}}

### Commands and control structures

> In the broadest sense, this refers to everything that is connected with control when variables are output.
> is connected.   

Reviewer: ```suggestion
In the broadest sense, this refers to everything that is connected to controlling the output of variables.
```
